packages/drivers/odsp-driver/src/odspDeltaStorageService.ts

```diff
@@ -49,62 +50,70 @@ export class OdspDeltaStorageService {
-				"Content-Type": `multipart/form-data;boundary=${formBoundary}`,
-			};
-
-			// Some request take a long time (1-2 minutes) to complete, where telemetry shows very small amount
-			// of time spent on server, and usually small payload sizes. I.e. all the time is spent somewhere in
-			// networking. Even bigger problem - a lot of requests timeout (based on cursory look - after 1-2 minutes)
-			// So adding some timeout to ensure we retry again in hope of faster success.
-			// Please see https://github.com/microsoft/FluidFramework/issues/6997 for details.
-			const abort = new AbortController();
-			const timer = setTimeout(() => abort.abort(), 30000);
-
-			const response = await this.epochTracker.fetchAndParseAsJSON<IDeltaStorageGetResponse>(
-				baseUrl,
+			return PerformanceEvent.timedExecAsync(
+				this.logger,
 				{
-					headers,
-					body: postBody,
-					method: "POST",
-					signal: abort.signal,
+					eventName: "OpsFetch",
+					attempts: options.refresh ? 2 : 1,
+					from,
+					to,
+					...telemetryProps,
+					reason: scenarioName,
+				},
+				async (event) => {
+					const formBoundary = uuid();
+					let postBody = `--${formBoundary}\r\n`;
+					postBody += `Authorization: Bearer ${storageToken}\r\n`;
+					postBody += `X-HTTP-Method-Override: GET\r\n`;
+
+					postBody += `_post: 1\r\n`;
+					postBody += `\r\n--${formBoundary}--`;
+					const headers: { [index: string]: any } = {
+						"Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+					};
+
+					// Some request take a long time (1-2 minutes) to complete, where telemetry shows very small amount
+					// of time spent on server, and usually small payload sizes. I.e. all the time is spent somewhere in
+					// networking. Even bigger problem - a lot of requests timeout (based on cursory look - after 1-2 minutes)
+					// So adding some timeout to ensure we retry again in hope of faster success.
+					// Please see https://github.com/microsoft/FluidFramework/issues/6997 for details.
+					const abort = new AbortController();
+					const timer = setTimeout(() => abort.abort(), 30000);
+
+					const response =
+						await this.epochTracker.fetchAndParseAsJSON<IDeltaStorageGetResponse>(
+							baseUrl,
+							{
+								headers,
+								body: postBody,
+								method: "POST",
+								signal: abort.signal,
+							},
+							"ops",
+							true,
+							scenarioName,
+						);
+					clearTimeout(timer);
+					const deltaStorageResponse = response.content;
+					const messages =
+						deltaStorageResponse.value.length > 0 &&
+						"op" in deltaStorageResponse.value[0]
+							? (deltaStorageResponse.value as ISequencedDeltaOpMessage[]).map(
+									(operation) => operation.op,
+							  )
+							: (deltaStorageResponse.value as ISequencedDocumentMessage[]);
+
+					event.end({
+						headers: Object.keys(headers).length !== 0 ? true : undefined,
+						length: messages.length,
+						...response.propsToLog,
```

Comment:
I think this will have duration property that will overwrite performance event property. That's probably fine, but worth checking for all REST calls (TreesLatest included) to report same metric, as they might differ slightly.